oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -0,0 +1,289 @@
+        :param percent_identity: maximum similarity between probes and target sequences, ranging from 0 to 100% (no missmatch)
+        :type percent_identity: int
+        :param coverage: minimum coverage between probes and target sequence, ranging from 0 to 100% (full coverage)
+        :type coverage: int
+        :param probe_length_min: minimum length of the probes created
+        :type probe_length_min: int
+        :param probe_length_max: maximum length of the probes created
+        :type probe_length_max: int
+        :param ligation_region: coverage between probes and target sequence should not span region around ligation site (e.g. ligation_region = 5 would correspond to -4 to +5 nt around ligation site), if ligation_region = 0, omit this requirement
+        :type ligation_region: int
+        """
+        super().__init__(n_jobs, dir_output, dir_annotations)
+
+        self.word_size = word_size
+        self.percent_identity = percent_identity
+        self.coverage = coverage
+        self.file_transcriptome_fasta = file_transcriptome_fasta
+        self.probe_length_min = probe_length_min
+        self.probe_length_max = probe_length_max
+        self.ligation_region = ligation_region
+
+        self.dir_blast = os.path.join(self.dir_output, "blast")
+        Path(self.dir_blast).mkdir(parents=True, exist_ok=True)
+
+        self.dir_probes = os.path.join(self.dir_output, "probes_blast")
+        Path(self.dir_probes).mkdir(parents=True, exist_ok=True)
+
+        self.file_removed_genes = os.path.join(
+            self.dir_probes, "genes_with_insufficient_probes.txt"
+        )
+
+    def run_blast_search(self):
+        """Run BlastN alignment tool to find regions of local similarity between sequences, where sequences are probes and transcripts.
+        BlastN identifies the transcript regions where probes match with a certain coverage and similarity.
+        """
+
+        def _run_blast(batch_id):
+            """Run BlastN alignment search for all probes of one batch
+            :param batch_id: Batch ID.
+            :type batch_id: int
+            """
+            for subbatch_id in range(self.number_subbatches):
+                file_probe_fasta_batch = os.path.join(
+                    self.dir_annotations,
+                    "probes_sequence_batch{}_{}.fna".format(batch_id, subbatch_id),
+                )
+                file_blast_batch = os.path.join(
+                    self.dir_blast, "blast_batch{}_{}.txt".format(batch_id, subbatch_id)
+                )
+
+                cmd = NcbiblastnCommandline(
+                    query=file_probe_fasta_batch,
+                    db=self.file_transcriptome_fasta,
+                    outfmt="10 qseqid sseqid length qstart qend qlen",
+                    out=file_blast_batch,
+                    strand="plus",
+                    word_size=self.word_size,
+                    perc_identity=self.percent_identity,
+                    num_threads=self.n_jobs,
```

Comment:
I'm not sure if we can have n_jobs in the blast command line call and also when calling this function... this would mean we will start n_job x n_jobs processes, no?